go-controller/pkg/ovn/ovn.go

```diff
@@ -225,84 +255,193 @@ func getPodNamespacedName(pod *kapi.Pod) string {
+	}
+	if err := libovsdbops.UpdateNBGlobalSetOptions(cm.nbClient, &nbGlobal); err != nil {
+		return fmt.Errorf("failed to set NB global option to enable logical datapath groups: %v", err)
+	}
+
+	metrics.RunTimestamp(cm.defaultStopChan, cm.sbClient, cm.nbClient)
+	metrics.MonitorIPSec(cm.nbClient)
+	if config.Metrics.EnableConfigDuration {
+		// with k=10,
+		//  for a cluster with 10 nodes, measurement of 1 in every 100 requests
+		//  for a cluster with 100 nodes, measurement of 1 in every 1000 requests
+		metrics.GetConfigDurationRecorder().Run(cm.nbClient, cm.kube, 10, time.Second*5, cm.defaultStopChan)
+	}
+
+	cm.podRecorder.Run(cm.sbClient, cm.defaultStopChan)
+
+	// Start and sync the watch factory to begin listening for events
+	if err := cm.watchFactory.Start(); err != nil {
+		return err
+	}
+	return nil
+}
+
 // NewOvnController creates a new OVN controller for creating logical network
 // infrastructure and policy
-func NewOvnController(ovnClient *util.OVNClientset, wf *factory.WatchFactory, stopChan <-chan struct{}, addressSetFactory addressset.AddressSetFactory,
-	libovsdbOvnNBClient libovsdbclient.Client, libovsdbOvnSBClient libovsdbclient.Client,
-	recorder record.EventRecorder) *Controller {
-	if addressSetFactory == nil {
-		addressSetFactory = addressset.NewOvnAddressSetFactory(libovsdbOvnNBClient)
-	}
-	svcController, svcFactory := newServiceController(ovnClient.KubeClient, libovsdbOvnNBClient, recorder)
-	egressSvcController := newEgressServiceController(ovnClient.KubeClient, libovsdbOvnNBClient, svcFactory, stopChan)
-	var hybridOverlaySubnetAllocator *subnetallocator.SubnetAllocator
-	if config.HybridOverlay.Enabled {
-		hybridOverlaySubnetAllocator = subnetallocator.NewSubnetAllocator()
-	}
-	return &Controller{
-		client: ovnClient.KubeClient,
-		kube: &kube.Kube{
-			KClient:              ovnClient.KubeClient,
-			EIPClient:            ovnClient.EgressIPClient,
-			EgressFirewallClient: ovnClient.EgressFirewallClient,
-			CloudNetworkClient:   ovnClient.CloudNetworkClient,
-		},
-		watchFactory:                 wf,
-		stopChan:                     stopChan,
-		masterSubnetAllocator:        subnetallocator.NewSubnetAllocator(),
-		hybridOverlaySubnetAllocator: hybridOverlaySubnetAllocator,
-		lsManager:                    lsm.NewLogicalSwitchManager(),
-		logicalPortCache:             newPortCache(stopChan),
-		namespaces:                   make(map[string]*namespaceInfo),
-		namespacesMutex:              sync.Mutex{},
-		externalGWCache:              make(map[ktypes.NamespacedName]*externalRouteInfo),
-		exGWCacheMutex:               sync.RWMutex{},
-		addressSetFactory:            addressSetFactory,
-		sharedNetpolPortGroups:       syncmap.NewSyncMap[*defaultDenyPortGroups](),
-		eIPC: egressIPController{
+func (cm *ControllerManager) NewOvnController(nadInfo *util.NetAttachDefInfo,
+	addressSetFactory addressset.AddressSetFactory) (*Controller, error) {
+	if nadInfo.NetCidr == "" {
```

Comment:
as discussed previously, we need to have a way (when using Flat L2) to optionally have OVN managed IP configuration - but tolerate when we don't. 

As such, I don't think we can have this condition.

Reply:
this will change after we introduce more types of secondary network toplogies. For now it should be good right. Also, I would think even for flat l2, we'd still need the cidr (per-cluster), and pod IP will be allocated from it.

Reply:
Why do you need the logical network to have a CIDR ? Maybe I'm missing something. 

Multus allows for L2 networks, and AFAIU, OVN also allows for L2 networks. We also need to have OVN provide L2 networks - and have something - other than OVN - manage the IPs on it. 

To clarify: I'm not saying this attribute is not needed: there are lots of use cases that will use it. I just don't think we can define it as **required** because there are also use cases whose IP addressing on the subnet is handled by some other entity.

Since this PR does **not** add any topology, nor enables the secondary networks to be actually created / used, I see no reason to define attributes as mandatory or not.